hackathon_site/dashboard/frontend/src/components/inventory/Item/Item.test.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import Item from "./Item";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+
+describe("InventoryItem", () => {
+    const image =
+        "hackathon_site/dashboard/frontend/src/assets/images/testImages/IO6e5a6.jpg";
+    const title = "Some Hardware";
+    const total = 6;
+    let limit = null;
+    let currentStock = 4;
+
+    test("Has stock with no limit", () => {
+        const { asFragment } = render(
+            <Item
+                image={image}
+                title={title}
+                total={total}
+                limit={limit}
+                currentStock={currentStock}
+            />
+        );
+        expect(asFragment()).toMatchSnapshot();
+    });
+
+    test("Out of Stock", () => {
+        let currentStock = null;
+        const { asFragment } = render(
+            <Item
+                image={image}
+                title={title}
+                total={total}
+                limit={limit}
+                currentStock={currentStock}
+            />
+        );
+        expect(asFragment()).toMatchSnapshot();
+    });
+});
```

Comment:
To test out of stock, I would use

```javascript
const { getByText } = render( blah );
expect(getByText("Some Hardware")).toBeInTheDocument();
expect(getByText("Out of stock")).toBeInTheDocument();
```

Same with testing if it has stock:
```javascript
const { getByText } = render( blah );
expect(getByText("Some Hardware")).toBeInTheDocument();
expect(getByText("4 of 6 in stock")).toBeInTheDocument();
```

I don't think a snapshot is needed, but @grahamhoyes please confirm.

Also a currentStock would more likely be 0 than null.

Reply:
There's also a `getByAltTest` to see if the image is there

Reply:
I would agree with @lisa-sa-li, I prefer explicit tests over snapshot because they are more clear about what is going on, and cause less clutter. I assume she is referring to `getByAltText` that you can use.